src/renderers/shared/reconciler/instantiateReactComponent.js

```diff
@@ -83,6 +106,8 @@ function instantiateReactComponent(node, parentCompositeType) {
       // represenations. I.e. ART. Once those are updated to use the string
       // representation, we can drop this code path.
       instance = new element.type(element);
+    } else if (isStatelessComponentType(element.type)) {
```

Comment:
@sebmarkbage I found a problem with that kind of components while trying to fix this: https://github.com/facebook/react/pull/4109